src/api/v1/db/oracledb/terms-dao.js

```diff
@@ -33,10 +49,37 @@ const getPostCurrentPreInterimTermCodes = async (connection) => {
 const getTerms = async (query) => {
   const connection = await getConnection();
   try {
-    const { rows } = await connection.execute(contrib.getTerms());
     const postCurrPreTermCodes = await getPostCurrentPreInterimTermCodes(connection);
-    const serializedTerms = serializeTerms(rows, postCurrPreTermCodes, query);
-    return serializedTerms;
+
+    const { postInterimTermDate, preInterimTermDate } = query;
+
+    const postInterimTermOfDate = await getInterimTermOfDate(
+      connection, postInterimTermDate, contrib.getPostInterimTermOfDate,
```

Comment:
postInterimTermDate and below preInterimTermDate could be undefined (or null), in which case we don't want to run these queries. Is that correct?

Reply:
I added a logic here to prevent unnecessary query if the input parameter is undefined: https://github.com/osu-mist/terms-api/blob/3b72fc75b717f06d9e878d8655ffbaeb884d578c/src/api/v1/db/oracledb/terms-dao.js#L36-L41